### PR TITLE
Add missing things to views

### LINKED ISSUE #119: add item description to cart

### LINKED ISSUE #121: on order show
- subtotal
- number to currency for total
- button on item # as opposed to 'view'
- status 'canceled'?

### LINKED ISSUE #123: fix retired button

### LINKED ISSUE #124: on admin orders
- change button from sort orders to filter
- fix order status table to display canceled
- fix link to admin_order_path
- link on admin order detail link to admin items not items

### PR DESCRIPTION
**Pull request by CjMoore**
- adds item description to cart
- fixes retired button on retired items
- adds subtotal to order show 
- adds method on item model to grab item subtotal
- adds number_to_currency on order show
- changes link to order show on orders index from 'view' to a button
- changes button on admin orders from 'sort orders' to filter
- fixes link on admin orders to direct to admin order
- styles admin orders and orders tables

closes #131 
closes #124 
closes #121 
closes #123 
closes #119

### CODE REVIEW

File: app/models/item.rb [hunk 1]
```diff
@@ -8,6 +8,14 @@ class Item < ApplicationRecord
...
-  	order_items.find_by(order_id: order.id).quantity
+  	get_order_item(order).quantity
+  end
+
+  def get_order_item(order)
+    order_items.find_by(order_id: order.id)
+  end
+
+  def get_subtotal_on_order(order)
+    get_order_item(order).subtotal
```
Comment (meyerhoferc) [inline app/models/item.rb:19]: 👍

File: app/views/admin/orders/index.html.erb [hunk 2]
```diff
@@ -13,8 +13,8 @@
       <td><%= @status["paid"] %></td>
     </tr>
     <tr>
-      <th>Cancelled</th>
-      <td><%= @status["cancelled"] %></td>
+      <th>Canceled</th>
+      <td><%= @status["canceled"] %></td>
```
Comment (meyerhoferc) [inline app/views/admin/orders/index.html.erb:17]: glad you found this

File: app/views/admin/orders/index.html.erb [hunk 3]
```diff
@@ -30,36 +30,40 @@
       <option value="" disabled selected>Choose your status</option>
       <option value = "ordered">Ordered</option>
       <option value = 'paid'>Paid</option>
-      <option value = 'canceled'>Cancelled</option>
+      <option value = 'canceled'>Canceled</option>
       <option value = 'completed'>Completed</option>
       <option value = 'all'>All Orders</option>
     </select></br>
-    <%= f.submit "sort orders", class: "waves-effect waves-light btn" %>
+    <%= f.submit "filter", class: "waves-effect waves-light btn" %>
```
Comment (meyerhoferc) [inline app/views/admin/orders/index.html.erb:37]: 👍

File: app/views/admin/orders/index.html.erb [hunk 4]
```diff
@@ -30,36 +30,40 @@
...
+          <% if order.status == "paid" %>
+              <%= button_to "Complete", admin_order_path(order, status: 'completed'), method: :put, class: "btn change-status" %>
+          <% end %>
+          <% if order.status == 'ordered' || order.status == 'paid' %>
+              <%= button_to "Cancel", admin_order_path(order, status: 'canceled'), method: :put, class: "btn change-status" %>
+          <% end %>
+          </td>
+        </tr>
+    <% end %>
+  </tbody>
```
Comment (meyerhoferc) [inline app/views/admin/orders/index.html.erb:68]: 👍

File: app/views/carts/show.html.erb [hunk 5]
```diff
@@ -8,25 +8,27 @@
...
       </thead>
 
       <tbody>
         <% @cart.items.each do |item, quantity| %>
         <tr>
           <td class="avatar"><img src="<%=item.image_tag%>" width=75px height=75px class="circle"/></td>
-          <td><h5><%= link_to item.title, item_path(item) %></h5></td>
-          <td><h5><%= quantity %></h5></td>
+          <td><%= link_to item.title, item_path(item) %></td>
+          <td class="truncate item-description"><%= item.description %></td>
```
Comment (meyerhoferc) [inline app/views/carts/show.html.erb:30]: this seems like an easy fix

File: app/views/items/show.html.erb [hunk 6]
```diff
@@ -18,9 +18,9 @@
             <li>Description: <%= @item.description %></li>
           </ul>
           <% if @item.available? %>
-            <%= button_to "Add to Cart", carts_path(item_id: @item.id) %>
+            <%= button_to "Add to Cart", carts_path(item_id: @item.id), class: "btn" %>
           <% else %>
-            <%= button_to "Retired", class: "btn disabled" %>
+            <button  class="btn disabled">Retired</button>
```
Comment (meyerhoferc) [inline app/views/items/show.html.erb:23]: much better!

File: app/views/orders/show.html.erb [hunk 7]
```diff
@@ -25,17 +28,17 @@
...
     </tr>
   </thead>
   <tbody>
     <tr>
-      <td><%= @order.total_price %></td>
+      <td><%= number_to_currency(@order.total_price, precision: 2) %></td>
       <td><%= @order.status %></td>
       <td><%= @order.created_at %></td>
-      <% if @order.status == "completed" || @order.status == "cancelled" %>
+      <% unless @order.status == "ordered"  %>
```
Comment (meyerhoferc) [inline app/views/orders/show.html.erb:41]: cleaner logic here

File: spec/features/orders/authenticated_user_can_checkout_spec.rb [hunk 8]
```diff
@@ -41,7 +41,7 @@
     expect(current_path).to eq(orders_path)
     # save_and_open_page
     within("table") do
-      expect(page).to have_content(Order.all.last.id)
+      expect(page).to have_button(Order.all.last.id)
```
Comment (meyerhoferc) [inline spec/features/orders/authenticated_user_can_checkout_spec.rb:44]: 👍

File: spec/features/orders/user_can_view_all_of_their_orders_spec.rb [hunk 9]
```diff
@@ -16,12 +16,11 @@
...
     visit orders_path
-    
-    expect(page).to have_link("View")
+
+    expect(page).to have_button(order1.id)
     expect(page).to have_content(order1.total_price)
-    expect(page).to have_content(order1.id.to_s)
     expect(page).to have_content(order1.total_price)
-    expect(page).to have_content(order2.id.to_s)
+    expect(page).to have_button(order2.id)
```
Comment (meyerhoferc) [inline spec/features/orders/user_can_view_all_of_their_orders_spec.rb:23]: 👍

**Review by meyerhoferc — APPROVED** [review]
Excellent fixes, love the attention to detail and fixes in the tests. Ready for second review + merge @valerietrudell @cmacaulay

File: app/assets/stylesheets/application.css [hunk 10]
```diff
@@ -66,6 +66,49 @@
...
+  }
+
+  .admin-orders .change-status {
+    margin: 5px 0 5px 0;
+  }
+
+  #view-admin-order {
+    margin-left: 10px;
+  }
+
```
Comment (cmacaulay) [inline app/assets/stylesheets/application.css:111]: looks good!

File: app/controllers/orders_controller.rb [hunk 11]
```diff
@@ -4,7 +4,7 @@ def index
   end
 
   def show
-    @order = current_user.orders.find_by(params[:order_id])
+    @order = current_user.orders.find(params[:id])
   end
```
Comment (cmacaulay) [inline app/controllers/orders_controller.rb:9]: 👍

File: app/models/item.rb (hunk 1 above)
Comment (cmacaulay) [reply app/models/item.rb:19]: 🎉

**Review by cmacaulay — APPROVED** [review]

**Pull request merged**